packages/terra-overlay/src/Overlay.jsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import FocusTrap from 'focus-trap-react';
+import 'terra-base/lib/baseStyles';
+import './Overlay.scss';
+
+const propTypes = {
+  /**
+  * The content to be displayed within the overlay.
+  */
+  children: PropTypes.node,
+  /**
+  * Incidates if the overlay is open.
+  */
+  isOpen: PropTypes.bool,
+  /**
+  * The visual theme to be applied to the overlay background. Accepts 'light', 'dark', and 'clear'.
+  */
+  backgroundStyle: PropTypes.oneOf(['light', 'dark', 'clear']),
+  /**
+  * Incidates if the overlay content is scrollable.
+  */
+  isScrollable: PropTypes.bool,
+  /**
+  * Callback triggered on overlay click or ESC key. Setting this enables close behavior.
+  */
+  onRequestClose: PropTypes.func,
+  /**
+  * Indicates if the overlay is relative to the triggering container.
+  */
+  isRelativeToContainer: PropTypes.bool,
+};
+
+const KEYCODES = {
+  ESCAPE: 27,
+};
+
+const defaultProps = {
+  children: null,
+  isOpen: false,
+  backgroundStyle: 'light',
+  isScrollable: false,
+  isRelativeToContainer: false,
+  onRequestClose: undefined,
+};
+
+class Overlay extends React.Component {
+  constructor(props) {
+    super(props);
+    this.setContainer = this.setContainer.bind(this);
+    this.disableContainerChildrenFocus = this.disableContainerChildrenFocus.bind(this);
+    this.enableContainerChildrenFocus = this.enableContainerChildrenFocus.bind(this);
+    this.shouldHandleESCKeydown = this.shouldHandleESCKeydown.bind(this);
+    this.shouldHandleClick = this.shouldHandleClick.bind(this);
+    this.resetBeforeOverlay = this.resetBeforeOverlay.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.shouldHandleESCKeydown);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.shouldHandleESCKeydown);
+    this.children = null;
+    this.container = null;
+  }
+
+  setContainer(node) {
+    if (!node) { return; } // Ref callbacks happen on mount and unmount, element is null on unmount
+    this.overflow = document.documentElement.style.overflow;
+
+    if (this.props.isRelativeToContainer) {
+      this.container = node.parentNode;
+      this.containerChildren = this.container.children;
+      this.disableContainerChildrenFocus();
+    } else {
+      document.documentElement.style.overflow = 'hidden';
+      this.container = null;
+    }
+  }
+
+  disableContainerChildrenFocus() {
+    if (this.containerChildren) {
+      const prevTabIndex = [];
+      for (let i = 0; i < this.containerChildren.length; i += 1) {
```

Comment:
Unfortunately no. I tried map, but this.containerChildren is an HTMLCollection and map only works on arrays. I looked into using a looping method and found this https://stackoverflow.com/questions/22754315/for-loop-for-htmlcollection-elements.